app/Model/Attribute.php

```diff
@@ -3909,6 +3908,74 @@ public function editAttribute($attribute, $eventId, $user, $objectId, $log = fal
         return true;
     }
 
+    public function __delete($id, $user, $hard = false)
+    {
+        $this->id = $id;
+        if (!$this->exists()) {
+            return false;
+        }
+        $result = $this->find('first', array(
+            'conditions' => array('Attribute.id' => $id),
+            'fields' => array('Attribute.*'),
```

Comment:
'recursive' => -1,
'contain' => array('Event')

would achieve the same and be cleaner.